node-api/backend/validator.go

```diff
@@ -21,11 +21,75 @@
 package backend
 
 import (
+	"slices"
+
 	"github.com/berachain/beacon-kit/node-api/backend/utils"
 	beacontypes "github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
 	"github.com/berachain/beacon-kit/primitives/math"
 )
 
+// FilteredValidators will grab all of the validators from the state at the
+// given slot. It will then filter them by the provided ids and statuses.
+func (b Backend) FilteredValidators(
+	slot math.Slot, ids []string, statuses []string,
+) ([]*beacontypes.ValidatorData, error) {
+	st, _, err := b.stateFromSlot(slot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert requested ids (can be validator index or pubkey) to validator index only.
+	validatorIndicies := make([]uint64, 0, len(ids))
+	for _, id := range ids {
+		validatorIndex, vErr := utils.ValidatorIndexByID(st, id)
+		if vErr != nil {
+			return nil, vErr
+		}
+		validatorIndicies = append(validatorIndicies, validatorIndex.Unwrap())
+	}
+
+	validators, err := st.GetValidators()
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter on validator indexes and statuses.
+	validatorData := make([]*beacontypes.ValidatorData, 0, len(validators))
+	for _, validator := range validators {
+		// Skip the validator if we are filtering by indicies and this validator is not included.
```

Comment:
type maybe? `indicies of this validator`

Reply:
Correct me if I'm wrong, but I think the only typo is `indicies` should be spelled `indices`. The comment is saying to skip this validator if 1) we filter by indices AND 2) this validator is not included.

Reply:
My understanding is same as Cal's. @shotes Could you pls confirm.

Reply:
Thanks Shotes, made changes. I have renamed the variable name to `validatorIndices`.